app/view_objects/find/results_view.rb

```diff
@@ -0,0 +1,260 @@
+module Find
+  # FIND:TODO Will be fleshed out in an upcoming ticket
+  class ResultsView
+    attr_reader :query_parameters
+
+    include ActionView::Helpers::NumberHelper
+
+    MAXIMUM_NUMBER_OF_SUBJECTS = 43
+    DISTANCE = "2".freeze
+    SUGGESTED_SEARCH_THRESHOLD = 3
+    MAXIMUM_NUMBER_OF_SUGGESTED_LINKS = 2
+    RESULTS_PER_PAGE = 10
+    MILES = "50".freeze
+
+    def initialize(query_parameters:)
+      @query_parameters = query_parameters
+    end
+
+    def query_parameters_with_defaults
+      query_parameters.except("utf8", "authenticity_token")
+        .merge(qualifications_parameters)
+        .merge(fulltime_parameters)
+        .merge(parttime_parameters)
+        .merge(hasvacancies_parameters)
+        .merge(sen_courses_parameters)
+        .merge(subject_parameters)
+    end
+
+    def courses
+      @courses ||= RecruitmentCycle.current.courses.includes(
+        :enrichments,
+        subjects: [:financial_incentive],
+        site_statuses: [:site],
+        provider: %i[recruitment_cycle ucas_preferences],
+      ).findable.page(query_parameters[:page] || 1)
+    end
+
+    def number_of_courses_string
+      case course_count
+      when 0
+        "No courses"
+      when 1
+        "1 course"
+      else
+        "#{number_with_delimiter(course_count)} courses"
+      end
+    end
+
+    def course_count
+      courses.count
+    end
+
+    def subjects
+      subject_codes.any? ? filtered_subjects : all_subjects
+    end
+
+    def qualifications_parameters
+      { "qualifications" => query_parameters["qualifications"].presence || %w[QtsOnly PgdePgceWithQts Other] }
+    end
+
+    def fulltime_parameters
+      { "fulltime" => fulltime? }
+    end
+
+    def parttime_parameters
+      { "parttime" => parttime? }
+    end
+
+    def hasvacancies_parameters
+      { "hasvacancies" => hasvacancies? }
+    end
+
+    def sen_courses_parameters
+      { "senCourses" => sen_courses? }
+    end
+
+    def fulltime?
+      return false if query_parameters["fulltime"].nil?
+
+      query_parameters["fulltime"] == "true"
+    end
+
+    def parttime?
+      return false if query_parameters["parttime"].nil?
+
+      query_parameters["parttime"] == "true"
+    end
+
+    def hasvacancies?
+      return true if query_parameters["hasvacancies"].nil?
+
+      query_parameters["hasvacancies"] == "true"
+    end
+
+    def sen_courses?
+      query_parameters["senCourses"] == "true"
+    end
+
+    # FIND:TODO double check
+    def subject_parameters
+      query_parameters["subject_codes"].present? ? { "subject_codes" => query_parameters["subject_codes"].presence } : {}
+    end
+
+    def subject_codes
+      query_parameters["subject_codes"] || []
+    end
+
+    def all_subjects
+      @all_subjects ||= Subject.select(:subject_name, :subject_code).order(:subject_name).all
```

Comment:
```suggestion
      @all_subjects ||= Subject.active.where.not(subject_code: nil).select(:subject_name, :subject_code).order(:subject_name).all
```

There is a few here that should not be... 

![image](https://user-images.githubusercontent.com/470137/198544040-79754459-2465-4363-8c83-a7b96b823cbc.png)


We have lost `humanities`!